File: client-app/WindowsClient/Models/StorageConfigModel.cs
namespace AI_BACKUP.WindowsService.Models
{
    public class StorageConfigModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public StorageType StorageType { get; set; }
        public bool IsDefault { get; set; }
        public bool IsEnabled { get; set; } = true;
        public string? ServerName { get; set; }
        public int Port { get; set; }
        public string? Username { get; set; }
        public string? ConnectionString { get; set; }
        public string? AccessKey { get; set; }
        public string? SecretKey { get; set; }
        public string? BucketName { get; set; }
        public string? ContainerName { get; set; }
        public string? FolderPath { get; set; }
        public string? Region { get; set; }
        public string? Endpoint { get; set; }
        public string? PrivateKeyPath { get; set; }
    }

    public enum StorageType
    {
        Local,
        GoogleDrive,
        OneDrive,
        AmazonS3,
        AzureBlob,
        FTP,
        SFTP
    }
}

File: client/src/app/models/storage-config.model.ts
export enum StorageType {
  Local = 'Local',
  GoogleDrive = 'GoogleDrive',
  OneDrive = 'OneDrive',
  AmazonS3 = 'AmazonS3',
  AzureBlob = 'AzureBlob',
  FTP = 'FTP',
  SFTP = 'SFTP'
}

export interface StorageConfig {
  id: number;
  userId: number;
  name: string;
  description?: string;
  storageType: StorageType;
  isDefault: boolean;
  connectionString?: string;
  accessKey?: string;
  secretKey?: string;
  bucketName?: string;
  containerName?: string;
  folderPath?: string;
  region?: string;
  endpoint?: string;
  createdAt: Date;
  updatedAt?: Date;
}